src/main/java/org/spongepowered/common/mixin/core/network/datasync/MixinEntityDataManager.java

```diff
@@ -76,12 +81,12 @@
         // Sponge Start - set up the current value, so we don't have to retrieve it multiple times
         final T currentValue = dataentry.getValue();
         final T incomingValue = value;
-        if (ObjectUtils.notEqual(value, currentValue)) { // Sponge - change dataentry.getValue() to use local variable
+        if (!value.equals(currentValue)) { // Sponge - change dataentry.getValue() to use local variable
             // Sponge Start - retrieve the associated key, if available
             // Client side can have an entity, because reasons.......
             // Really silly reasons......
             // I don't know, ask Grum....
-            if (this.entity != null && this.entity.world != null && !this.entity.world.isRemote) { // We only want to spam the server world ;)
+            if (ShouldFire.CHANGE_DATA_HOLDER_EVENT_VALUE_CHANGE && this.entity != null && this.entity.world != null && !this.entity.world.isRemote) { // We only want to spam the server world ;)
```

Comment:
We should only do the code within this method if this is true. Otherwise skip all of this to be as fast as the engine was.

Reply:
This is an Overwrite.

Even if the event is not fired, what is after //Sponge End need to be executed